File: flatdata-rs/tests/features/src/constants/values.rs
#![allow(dead_code)]

include!(concat!(env!("OUT_DIR"), "/constants/values.rs"));

#[test]
fn values() {
    assert_eq!(n::FOO_I8_NEG, -128);
    assert_eq!(n::FOO_I8_POS, 127);
    assert_eq!(n::FOO_I8_ZERO, 0);
    assert_eq!(n::FOO_I8_NEG_HEX, -0x80);
    assert_eq!(n::FOO_I8_POS_HEX, 0x7f);
    assert_eq!(n::FOO_I8_ZERO_HEX, 0x0);
    assert_eq!(n::FOO_U8_POS, 255);
    assert_eq!(n::FOO_U8_ZERO, 0);
    assert_eq!(n::FOO_U8_POS_HEX, 0xff);
    assert_eq!(n::FOO_U8_ZERO_HEX, 0x0);

    assert_eq!(n::FOO_I16_NEG, -32768);
    assert_eq!(n::FOO_I16_POS, 32767);
    assert_eq!(n::FOO_I16_ZERO, 0);
    assert_eq!(n::FOO_I16_NEG_HEX, -0x8000);
    assert_eq!(n::FOO_I16_POS_HEX, 0x7fff);
    assert_eq!(n::FOO_I16_ZERO_HEX, 0x0);
    assert_eq!(n::FOO_U16_POS, 65535);
    assert_eq!(n::FOO_U16_ZERO, 0);
    assert_eq!(n::FOO_U16_POS_HEX, 0xffff);
    assert_eq!(n::FOO_U16_ZERO_HEX, 0x0);

    assert_eq!(n::FOO_I32_NEG, -2147483648);
    assert_eq!(n::FOO_I32_POS, 2147483647);
    assert_eq!(n::FOO_I32_ZERO, 0);
    assert_eq!(n::FOO_I32_NEG_HEX, -0x80000000);
    assert_eq!(n::FOO_I32_POS_HEX, 0x7fffffff);
    assert_eq!(n::FOO_I32_ZERO_HEX, 0x0);
    assert_eq!(n::FOO_U32_POS, 4294967295);
    assert_eq!(n::FOO_U32_ZERO, 0);
    assert_eq!(n::FOO_U32_POS_HEX, 0xffffffff);
    assert_eq!(n::FOO_U32_ZERO_HEX, 0x0);

    assert_eq!(n::FOO_I64_NEG, -9223372036854775808);
    assert_eq!(n::FOO_I64_POS, 9223372036854775807);
    assert_eq!(n::FOO_I64_ZERO, 0);
    assert_eq!(n::FOO_I64_NEG_HEX, -0x8000000000000000);
    assert_eq!(n::FOO_I64_POS_HEX, 0x7fffffffffffffff);
    assert_eq!(n::FOO_I64_ZERO_HEX, 0x0);
    assert_eq!(n::FOO_U64_POS, 18446744073709551615);
    assert_eq!(n::FOO_U64_ZERO, 0);
    assert_eq!(n::FOO_U64_POS_HEX, 0xffffffffffffffff);
    assert_eq!(n::FOO_U64_ZERO_HEX, 0x0);
}

File: flatdata-generator/tests/generators/go_expectations/constants/values.go
package n

import (
	"bytes"
	"encoding/binary"
	"errors"
	"fmt"
	"log"

	"github.com/heremaps/flatdata/flatdata-go/flatdata"
)

const (
    flatdataOffsetSizeInBytes uint = 8
    flatdataPaddingSizeInBytes uint = 8
    FooI8Neg int8 = -128
    FooI8Pos int8 = 127
    FooI8Zero int8 = 0
    FooI8NegHex int8 = -128
    FooI8PosHex int8 = 127
    FooI8ZeroHex int8 = 0
    FooU8Pos uint8 = 255
    FooU8Zero uint8 = 0
    FooU8PosHex uint8 = 255
    FooU8ZeroHex uint8 = 0
    FooI16Neg int16 = -32768
    FooI16Pos int16 = 32767
    FooI16Zero int16 = 0
    FooI16NegHex int16 = -32768
    FooI16PosHex int16 = 32767
    FooI16ZeroHex int16 = 0
    FooU16Pos uint16 = 65535
    FooU16Zero uint16 = 0
    FooU16PosHex uint16 = 65535
    FooU16ZeroHex uint16 = 0
    FooI32Neg int32 = -2147483648
    FooI32Pos int32 = 2147483647
    FooI32Zero int32 = 0
    FooI32NegHex int32 = -2147483648
    FooI32PosHex int32 = 2147483647
    FooI32ZeroHex int32 = 0
    FooU32Pos uint32 = 4294967295
    FooU32Zero uint32 = 0
    FooU32PosHex uint32 = 4294967295
    FooU32ZeroHex uint32 = 0
    FooI64Neg int64 = -9223372036854775808
    FooI64Pos int64 = 9223372036854775807
    FooI64Zero int64 = 0
    FooI64NegHex int64 = -9223372036854775808
    FooI64PosHex int64 = 9223372036854775807
    FooI64ZeroHex int64 = 0
    FooU64Pos uint64 = 18446744073709551615
    FooU64Zero uint64 = 0
    FooU64PosHex uint64 = 18446744073709551615
    FooU64ZeroHex uint64 = 0
)
